lib/Mojo/Asset/Memory.pm

```diff
@@ -41,12 +41,21 @@ sub get_chunk {
 
 sub move_to { path($_[1])->spurt($_[0]{content} // '') and return $_[0] }
 
-sub size { length(shift->{content} // '') }
+sub size {
+  my $self = shift;
+  return length($self->{content} // '') unless $self->is_range;
+  return $self->_end_range(-1) - $self->start_range + 1;
+}
 
 sub slurp { shift->{content} // '' }
 
 sub to_file { Mojo::Asset::File->new->add_chunk(shift->slurp) }
 
+sub _end_range {
+  my $self = shift;
+  return undef unless $self->is_range;
+  return $self->end_range || shift() + length($self->{content} // '');
```

Comment:
Might be a personal thing, but having shift() there made me very confused.

Reply:
I think `shift()` there is bad style, unless it's absolutely necessary for performance.